hack/update-mocks.sh

```diff
@@ -44,10 +44,6 @@ cd "${KUBE_ROOT}"
 
 GENERATED_MOCK_FILE_REGEX="^// Code generated by mockery v[0-9.]\+. DO NOT EDIT.$"
 
-# pick a tempfile path for writing to
```

Comment:
Was the headers the only think we had depending on this? (I think so but we should confirm since there's an observable behavior difference to writing a temp file and then `mv` on the same filesystem versus writing directly to the target path)

Reply:
confirmed that verify-mocks.sh is operating using git worktree and doesn't depend on this.